scripts/update-to-latest.sh

```diff
@@ -20,33 +15,28 @@ git submodule update --init
 # always work with the latest inspector_protocol repo
 git submodule foreach git pull origin main
 
-# => cd into chromium
-cd "$chromium_src_path" || exit 1
-
-# get latest from chromium main
-git fetch origin main
-git checkout -f origin/main
-gclient sync --nohooks --delete_unversioned_trees --reset --force # no hooks needed.
-
-python_bin="$chromium_src_path/third_party/depot_tools/vpython3"
+python_bin="python3"
 
-browser_protocol_path="$chromium_src_path/third_party/blink/public/devtools_protocol/browser_protocol.pdl"
-js_protocol_path="$chromium_src_path/v8/include/js_protocol.pdl"
+chromium_deps_url="https://chromium.googlesource.com/chromium/src.git/+/refs/heads/main/DEPS?format=TEXT"
+v8_revision=$(curl --silent "${chromium_deps_url}" | base64 --decode | grep "'v8_revision':" | cut -d "'" -f4)
+browser_protocol_url="https://chromium.googlesource.com/chromium/src.git/+/refs/heads/main/third_party/blink/public/devtools_protocol/browser_protocol.pdl?format=TEXT"
+js_protocol_url="https://chromium.googlesource.com/v8/v8.git/+/${v8_revision}/include/js_protocol.pdl?format=TEXT"
 
-# copy the two protocol.pdl files over.
-cp "$js_protocol_path" "$protocol_repo_path/pdl"
-cp "$browser_protocol_path" "$protocol_repo_path/pdl"
+curl --silent "${browser_protocol_url}" | base64 --decode > pdl/browser_protocol.pdl
+curl --silent "${js_protocol_url}" | base64 --decode > pdl/js_protocol.pdl
 
 # extract cr revision number
-commit_pos_line=$(git log --date=iso --no-color --max-count=1 | gtac | grep -E -o "Cr-Commit-Position.*" | head -n1)
-commit_rev=$(echo "$commit_pos_line" | grep -E -o "\d+")
+chromium_git_log_url="https://chromium.googlesource.com/chromium/src.git/+log/refs/heads/main?format=JSON"
+commit_rev=$(curl --silent "${chromium_git_log_url}" | tail -n +2 | \
+  "$python_bin" -c 'import json,sys;print(json.load(sys.stdin)["log"][0]["message"])' | \
+  grep -E -o "^Cr-Commit-Position.*" | grep -E -o "\d+")
```

Comment:
i'm not 100% sure but i think i use `tac` to get the 'real' line when we have revert/reland commits which have 'cr-commit-position' multiple times in the message. i guess the `^` regex bit will do the same thing, assuming those nested messages always have the `>`. I'm fine with that assumption

Reply:
Yeah, that’s what I was thinking re: the `^`. I’m similarly unsure if this is actually enforced by Gerrit or whatever but it seemed reasonable to assume there’ll only be one `^Cr-Commit-Position` per commit message.